### ISSUE TITLE
NullReferenceException disposing TorControlClient

### ISSUE THREAD

**Issue by AbeniMatteo** [joelverhagen/TorSharp#35]
The following snippet throws `NullReferenceException` on `TorControlClient.Dispose`.
The internal `TcpClient` is initialized and assigned but the reader and writer are not.

```csharp
namespace Repro
{
    using Knapcode.TorSharp.Tools.Tor;

    class Program
    {
        static void Main()
        {
            using (var client = new TorControlClient())
            {
                try
                {
                    // Invalid ip
                    client.ConnectAsync("300.0.0.0", 12345).Wait();
                }
                catch
                {
                }
            }
        }
    }
}
```

**Comment by joelverhagen** [comment]
Nice find. Thanks for the repro too! I'll release a new version with the fix.

**Comment by joelverhagen** [comment]
Released [1.9.1](https://www.nuget.org/packages/Knapcode.TorSharp/1.9.1).

**Comment by AbeniMatteo** [comment]
Thank you.
